lib/crawly/middlewares/domain_filter.ex

```diff
@@ -19,8 +19,9 @@ defmodule Crawly.Middlewares.DomainFilter do
 
   def run(request, state, _opts \\ []) do
     base_url = state.spider_name.base_url()
+    parsed_url = URI.parse(request.url)
 
-    case String.contains?(request.url, base_url) do
+    case String.contains?(base_url, parsed_url.host) do
```

Comment:
This will raise errors when `parsed_url.host` is `nil`, such as when the request url is relative. Perhaps a `nil` check would be good.

Not too sure about the expected behaviour for requests built with relative urls. I would think that it would skip the domain filter entirely.

Reply:
Ok, I have added a check. Just in case.